### PR TITLE
BI-9193 Add llp and lp feature flags

### PR DESCRIPTION
**Pull request by cgregoryCH**
- Use feature flags to toggle validation for LP / LLP journeys

[BI-9193](https://companieshouse.atlassian.net/browse/BI-9193)

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/certificates/orders/api/validator/CertificateOptionsValidatorConfig.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
+package uk.gov.companieshouse.certificates.orders.api.validator;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import uk.gov.companieshouse.certificates.orders.api.config.FeatureOptions;
+
+import java.util.function.Consumer;
+
+@Configuration
```
Comment (kinpang-CH) [inline src/main/java/uk/gov/companieshouse/certificates/orders/api/validator/CertificateOptionsValidatorConfig.java:9]: Add JavaDocs

File: src/main/java/uk/gov/companieshouse/certificates/orders/api/validator/CertificateOptionsValidator.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
+package uk.gov.companieshouse.certificates.orders.api.validator;
+
+import uk.gov.companieshouse.certificates.orders.api.model.CertificateItemOptions;
+
+import java.util.List;
+import java.util.function.Consumer;
+
```
Comment (kinpang-CH) [inline src/main/java/uk/gov/companieshouse/certificates/orders/api/validator/CertificateOptionsValidator.java:7]: Add JavaDocs

File: src/main/java/uk/gov/companieshouse/certificates/orders/api/validator/OptionsValidationHelper.java [hunk 3]
```diff
@@ -0,0 +1,108 @@
+package uk.gov.companieshouse.certificates.orders.api.validator;
+
+import uk.gov.companieshouse.certificates.orders.api.model.CertificateItemOptions;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+class OptionsValidationHelper {
+    private final  CertificateItemOptions options;
```
Comment (kinpang-CH) [inline src/main/java/uk/gov/companieshouse/certificates/orders/api/validator/OptionsValidationHelper.java:10]: Stray space

**Review by kinpang-CH — COMMENTED** [review]
Address pull request comments

File: src/main/java/uk/gov/companieshouse/certificates/orders/api/validator/CertificateOptionsValidatorConfig.java (hunk 1 above)
Comment (phudson-ch) [reply src/main/java/uk/gov/companieshouse/certificates/orders/api/validator/CertificateOptionsValidatorConfig.java:9]: Javadocs added

File: src/main/java/uk/gov/companieshouse/certificates/orders/api/validator/CompanyType.java [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+public enum CompanyType {
+    LIMITED_PARTNERSHIP("limited-partnership"),
+    LIMITED_LIABILITY_PARTNERSHIP("llp");
+
+    private final String type;
+
+    CompanyType(String type) {
+        this.type = type;
+    }
+
```
[the remark this answers is not in the document]
Comment (phudson-ch) [reply src/main/java/uk/gov/companieshouse/certificates/orders/api/validator/CompanyType.java:12]: CompanyType.getEnumValue() implements

File: src/main/java/uk/gov/companieshouse/certificates/orders/api/validator/CertificateOptionsValidator.java (hunk 2 above)
Comment (phudson-ch) [reply src/main/java/uk/gov/companieshouse/certificates/orders/api/validator/CertificateOptionsValidator.java:7]: Javadocs added

**Review by kinpang-CH — APPROVED** [review]

**Review by kinpang-CH — APPROVED** [review]

**Pull request merged**